### PR TITLE
Final project adjustments

### PR DESCRIPTION
**Pull request by themagicianking**
Issue: The existing requirement for participants creating their [project timeline during prep week](https://github.com/Techtonica/curriculum/blob/main/projects/final-project/planning-your-project.md) does not ask participants to incorporate the milestones laid out for each week's milestones.

Feedback request: ensuring that I'm following general PR etiquette; I'd love to know how I can improve and anything I may have missed.

I have completed the following changes:

- [x] Added a timeline section to project planning that contains the weeks 1-4 milestones for reference
- [x] Added a prerequisite section to the final project main readme containing the final project crash course
- [x] Updated the link to the project template in the final project main readme 
- [x] Added details in [the prep google doc](https://docs.google.com/document/d/1NG41QXIa9C-q4-FMWcQi9g5Ncb3kffU3VwcM7Kw7Rvg/edit?usp=sharing), that were missing in the existing topic outline to  [the prep topic outline](https://github.com/Techtonica/curriculum/blob/main/projects/final-project/planning-your-project.md) and [the final project main topic outline prep section](https://github.com/Techtonica/curriculum/tree/main/projects/final-project#project-prep-week-milestones)
- [x] Added a prerequisite section in [the main final project](https://github.com/Techtonica/curriculum/tree/main/projects/final-project) exposing participants to [Final Project Crash Course](https://docs.google.com/presentation/d/1xbJ6KXlMo-9VsY2dU0qSpWdXa4eSwsHYD9aB1usk9R4/edit?usp=sharing)
- [x] Add a section after week 1 - 4 that clarify that ["GitHub repository requirements for code and PR Collaboration"](https://docs.google.com/document/d/1CcIJyAf9h9-Tua5TGsoaxwsQxhbW9tc5gHP0I2YMArY/edit?usp=sharing) are weekly requirements

To do:
- [ ] Reference work from the incoming issues creating topic outlines that may be relevant to point/link to in your updates (be sure to include comments in this issue and the following about how you have made them related)
- [ ] Reference incoming https://github.com/Techtonica/curriculum/issues/2106
- [ ] Reference incoming https://github.com/Techtonica/curriculum/issues/2107

I am waiting on [render deployment topic outline](https://github.com/Techtonica/curriculum/issues/2106) and [production topic outline](https://github.com/Techtonica/curriculum/issues/2107) to complete the relevant to do list items.

### CODE REVIEW

File: projects/final-project/planning-your-project.md [hunk 1]
```diff
@@ -94,6 +95,46 @@ Look at conventions on other sites and stick with them. For example, login links
...
+- [ ] Add setup instructions to your repo's README, and make any updates so it reflects the current state of the app. Don't forget a screenshot!
+- [ ] Have 1 code review and PR with your mentor and 1 PR partner.
+- [ ] Ensure you have at least 50 commits (closer to 200 is better).
+
+**Week 4 Milestones**
+- [ ] Present final project to cohort!
+- [ ] Record and submit your personal pitch and demo video—it should be 2 minute MAX for each part with NO READING
+
+- [ ] Create a daily project schedule based on the weeks 1-4 outlines. Week 4 can be used for stretch goals if time allows.
+
```
Comment (daaimah123) [inline projects/final-project/planning-your-project.md:137]: Let's remove the list of milestones here and directly refer to the Schedule section of [the main README.md](https://github.com/Techtonica/curriculum/blob/main/projects/final-project/README.md#schedule)

If there are any points in these milestones that were not in the main README, please update those lists

**Review by daaimah123 — CHANGES_REQUESTED** [review]
This is a great first pass. 🚀 You are missing a few requirements from [your PR requirement](https://docs.google.com/document/d/1Yx2S5xgnpJZmZRu5MCkDtbIcLnmOg84Rr5PpfQCqlRo/edit?usp=sharing):
- Reference the issue in the “Development” section on the right hand pane of your PR (this will enable both the issue and your PR to be connected in your Kanban board and automatically trigger processes such as closing the issue at the same time your PR is merged in)
- Assign the project of your PR to match the kanban board you are using during your sprint. 
  - Change the status of your issue to “In Review / In progress”, “Testing”, or “Implemented” where applicable (an open PR should not have “No status” under the projects section (see image)
- Assign yourself as the PR “Assignee”

Ideally you want the text where your PR lives to reflect where you are at in the PR. What this means is that when your PR is ready for review, some of your description and objective text will need to update accordingly so that it reflects your PR work that the time of being ready for review (some of your summary is reflective on in draft mode):
- Description: what is your issue that you've been assigned (context), later down the line passing along instructions for how to set up and test your PR code
- Objective: what is your PR doing and how does it provide a solution for your issue
how does the code in your PR hope to accomplish
what are your changes doing

Best Practice: give a slack nudge to your reviewer to let them know that you have request them as a reviewer on your PR

**Comment by mai-repo** [comment]
As Daaimah mentioned earlier, there are a few things to address, including the assignee, projects, and development section. It would help to provide more clarity in your PR description, along with an objective, but I believe Daaimah has already pointed out these issues. 

Overall, this is a great first draft! I really appreciate the work you've put in so far. Once you've seen this comment, please add an emoji to acknowledge it, and then leave a comment before resolving it!

**Review by mai-repo — APPROVED** [review]

File: projects/final-project/planning-your-project.md (hunk 1 above)
Comment (themagicianking) [reply projects/final-project/planning-your-project.md:103]: Removed the list and linked to the schedule section!

File: projects/final-project/planning-your-project.md [hunk 2]
```diff
@@ -97,43 +97,9 @@ Look at conventions on other sites and stick with them. For example, login links
...
-- [ ] Ensure you have at least 50 commits (closer to 200 is better).
-
-**Week 4 Milestones**
-- [ ] Present final project to cohort!
-- [ ] Record and submit your personal pitch and demo video—it should be 2 minute MAX for each part with NO READING
-
-- [ ] Create a daily project schedule based on the weeks 1-4 outlines. Week 4 can be used for stretch goals if time allows.
+Take a look at the suggested daily schedule and required milestones for all four weeks of your project referenced [here](https://github.com/Techtonica/curriculum/blob/main/projects/final-project/README.md#schedule).
+
+- [ ] Create a daily schedule for your project.
```
Comment (daaimah123) [inline projects/final-project/planning-your-project.md:102]:
```suggestion
Take a look at the suggested daily schedule and required milestones for all four weeks of your project referenced [here](https://github.com/Techtonica/curriculum/blob/main/projects/final-project/README.md#schedule). Create a daily schedule for your project.
```

With line 102 being a single task, lets add it to your call to action statement above it.

**Review by daaimah123 — CHANGES_REQUESTED** [review]

File: projects/final-project/planning-your-project.md (hunk 1 above)
Comment (daaimah123) [reply projects/final-project/planning-your-project.md:103]: If any of the removed milestone content that was previously in this PR was not referenced in the milestones that lives in the main README (for example perhaps the google doc milestones say something different from the README by having an extra bulleted requirement) than please ensure that the main README where the milestones will ultimately live are all inclusive and not missing anything from the google doc.

Comment (themagicianking) [reply projects/final-project/planning-your-project.md:103]: Thanks for the clarification! I have added the things I observed were missing from the main readme.

**Review by daaimah123 — CHANGES_REQUESTED** [review]
Replace template link with https://github.com/Techtonica/curriculum/blob/main/projects/2023TemplateWithVite/README.md

File: pair-programming/week-7/react-express-app/starter-code/React-express-starter-code/.gitignore [hunk 3]
```diff
@@ -1 +1,2 @@
 node_modules
+package-lock.json
```
Comment (ndp) [inline pair-programming/week-7/react-express-app/starter-code/React-express-starter-code/.gitignore:2]: Is this intentional? Generally you want to check this file in to strive for a consistent environment.

**Comment by ndp** [comment]
Looks fine to me (except the package.lock files)... but... I have no idea what the project was so I can't evaluate whether it is satisfied.

I usually make my own PR template that has these components:

1. **Project Description**. This might be a lengthy description about the goal of the PR, or it may just be a ticket number, depending on the environment.
2. **Technical approach** or notes. This is where you might including testing notes, if appropriate.
3. **Feedback requested**. I don't see this much, but I feel like it is critical. What do you want your reviewer to focus on? I have found that many times you get irrelevant or unactionable feedback, which then just causes you more work. If you can cut this off before it happens you'll have a much happier PR cycle. For example, the `package.lock` comment above may have been hashed out and misguided... but I didn't know.
4. **To Do**s. I always have a section so that I can start my PR review cycle before everything is done. You can edit the todo list and check things off as the PR gets ready for release. Github will track todos (`- [ ] do this`) found in individual comments, even if they are associated with certain lines in the PR, and you may also end up using that. But again, I put my todos in before I ask for reviews, so that people understand where I'm at.

Most companies will also have their own template, of varying qualities... but these are things I always try to include.

The other thing I do, as I open my PR, is give it a review myself. This usually results in me leaving some PR comments about specific changes-- again, guiding the feedback. Or perhaps expressing doubt, or asking for help. Lots of stuff... but I find there are lots of comments that don't belong as code comments, but are useful to say...

Anyway, I've said enough.

File: pair-programming/week-7/react-express-app/starter-code/React-express-starter-code/.gitignore (hunk 3 above)
Comment (daaimah123) [reply pair-programming/week-7/react-express-app/starter-code/React-express-starter-code/.gitignore:2]: woops, that's on me I was tinkering with the dependabot PRs and wanted to temporarily remove the package-lock files from the repo, this has been corrected on the main branch and should be resolved in the PR once the branch is updated with main

Comment (themagicianking) [reply pair-programming/week-7/react-express-app/starter-code/React-express-starter-code/.gitignore:2]: Thanks Andy, thanks Daaimah--just wanted to let you all know that this was seen :) I am now updating the pr.

Comment (daaimah123) [reply pair-programming/week-7/react-express-app/starter-code/React-express-starter-code/.gitignore:2]:
```suggestion
```

File: pair-programming/week-7/react-express-app/starter-code/React-express-starter-code/client/.gitignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+node_modules
+package-lock.json
```
Comment (daaimah123) [inline pair-programming/week-7/react-express-app/starter-code/React-express-starter-code/client/.gitignore:2]:
```suggestion
```

File: project-example/week-10/sample-code/.gitignore [hunk 5]
```diff
@@ -14,4 +14,6 @@
 .env.local
 .env.development.local
 .env.test.local
-.env.production.local
\ No newline at end of file
+.env.production.local
+
+package-lock.json
```
Comment (daaimah123) [inline project-example/week-10/sample-code/.gitignore:19]:
```suggestion
```

File: project-example/week-10/sample-code/client/.gitignore [hunk 6]
```diff
@@ -21,3 +21,5 @@
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
+
+package-lock.json
```
Comment (daaimah123) [inline project-example/week-10/sample-code/client/.gitignore:25]:
```suggestion
```

File: project-example/week-7/weather-forecast/.gitignore [hunk 7]
```diff
@@ -1,2 +1,3 @@
 node_modules
 .env
+package-lock.json
```
Comment (daaimah123) [inline project-example/week-7/weather-forecast/.gitignore:3]:
```suggestion
```

File: project-example/week-7/weather-forecast/client/.gitignore [hunk 8]
```diff
@@ -0,0 +1,2 @@
+node_modules
+package-lock.json
```
Comment (daaimah123) [inline project-example/week-7/weather-forecast/client/.gitignore:2]:
```suggestion
```

File: project-example/week-9/eventonica/eventonica-sample-code/client/.gitignore [hunk 9]
```diff
@@ -21,3 +21,5 @@
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
+
+package-lock.json
```
Comment (daaimah123) [inline project-example/week-9/eventonica/eventonica-sample-code/client/.gitignore:25]:
```suggestion
```

File: projects/2023TemplateWithVite/client/.gitignore [hunk 10]
```diff
@@ -24,3 +24,5 @@ dist-ssr
 *.njsproj
 *.sln
 *.sw?
+
+package-lock.json
```
Comment (daaimah123) [inline projects/2023TemplateWithVite/client/.gitignore:28]:
```suggestion
```

File: projects/2023TemplateWithVite/server/.gitignore [hunk 11]
```diff
@@ -21,3 +21,5 @@ client/node_modules/
 server/.env
 .DS_Store
 server/.DS_Store
+
+package-lock.json
```
Comment (daaimah123) [inline projects/2023TemplateWithVite/server/.gitignore:25]:
```suggestion
```

**Review by daaimah123 — CHANGES_REQUESTED** [review]
these changes will fix the conflicts for the `.gitignore` files

**Review by daaimah123 — APPROVED** [review]

**Pull request merged**